test/mux.sh

```diff
@@ -237,15 +243,26 @@ write_and_verify_logs() {
 }
 
 reset_ES_HOST() {
-    oc set env dc logging-mux $1 $2
+    muxpod=$( get_running_pod mux )
+    os::cmd::try_until_success "oc set env dc logging-mux $1 $2"
     os::cmd::try_until_failure "oc get pod $muxpod"
+    os::log::debug $( oc get pods -l component=mux )
     os::cmd::try_until_text "oc get pods -l component=mux" "^logging-mux-.* Running "
-    muxpod=$( get_running_pod mux )
```

Comment:
Why get rid of setting `muxpod` here?

Reply:
Well, the issue is after resetting ES_HOST and OPS_HOST, if the test logs are issued without making sure the new mux pod is 100% up and running by [1], the fluentd client fails to send the data and they are lost (since the current fluentd client are not using the file buffer [2].)  muxpod have to be retrieved after the code [1] is done.  If the way how to confirm the new mux pod's full availability [1] is not correct, I'm open to your suggestion.
```
[1]
    # wait for the file buffer disappears once
    os::cmd::try_until_failure 'oc get pods | egrep "logging-mux-.*-deploy"'
    os::cmd::try_until_failure 'oc get pods | egrep "logging-mux-.*Terminating"'
    os::cmd::try_until_text "oc get pods -l component=mux" "Running"
```
[2] https://github.com/openshift/origin-aggregated-logging/blob/master/fluentd/configs.d/filter-pre-mux-client.conf

Reply:
ok - I think the following should be sufficient:

    # do something to cause mux to be redeployed
    oc rollout status -w dc/logging-mux # wait for mux to be redeployed
    os::cmd::try_until_text "oc get pods -l component=mux" "Running"

After this, there still may be a mux pod in "Terminating" for a short period of time while the new one is in "Running", but that should be ok